src/eddymotion/model/dipy.py

```diff
@@ -330,6 +332,55 @@ def compute_spherical_covariance(
     return np.where(theta <= a, 1 - 3 * (theta / a) ** 2 + 2 * (theta / a) ** 3, 0)
 
 
+def compute_derivative(
+    theta: np.ndarray,
+    kernel: np.ndarray,
+    weighting: str,
+    params: dict[float],
+):
+    """
+    Compute the analytical derivative of the kernel.
+
+    Parameters
+    ----------
+    theta : :obj:`~numpy.ndarray`
+        Pairwise angles across diffusion gradient encoding directions.
+    kernel : :obj:`~numpy.ndarray`
+        Current kernel.
+    weighting : :obj:`str`
+        The kind of kernel which derivatives will be calculated.
+    params : :obj:`dict`
+        Current parameter values.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Gradients of the kernel.
+
+    """
+
+    n_partials = len(params)
+
+    a = params.pop("a")
+    lambda_s = params.pop("lambda_s")
+
+    min_angles = theta > a
+
+    if weighting == "spherical":
+        deriv_a = (3 * theta[min_angles] / a**2) - (1.5 * (theta[min_angles] / a) ** 2) / a
+    elif weighting == "exponential":
+        deriv_a = theta[min_angles] * a * np.exp(-theta[min_angles] / a)
```

Comment:
Missing a negative sign here and a should not be a multiplying factor ?

Reply:
Actually:

$$
\frac{\partial}{\partial x} e^{f(x)} = \frac{\partial f(x)}{\partial x} \cdot e^{f(x)}
$$

Hence:

$$
\frac{\partial}{\partial a} e^{- \frac{\theta}{a}} = \left( - \theta \frac{\partial a^{-1}}{\partial a} \right) \cdot e^{- \frac{\theta}{a}} = \frac{\theta}{a^2} \cdot e^{- \frac{\theta}{a}}
$$

```suggestion
        deriv_a = np.exp(-theta[min_angles] / a) * theta[min_angles]  / (a ** 2)
```

Correct?

Reply:
OK, had mistakenly took $\theta$ as the variable here. Thanks.